tests/test_dynamic_links.py

```diff
@@ -72,61 +73,68 @@ def teardown_module():
 
 
 class TestGetStats(object):
-    def test_get_stats(self, dltest):
-        dltest._instrument_dynamic_links(payload=MOCK_GET_STATS_RESPONSE)
+    def test_get_stats(self, dynamic_links_test):
+        _, recorder = dynamic_links_test._instrument_dynamic_links(
+            payload=MOCK_GET_STATS_RESPONSE)
         options = dynamic_links.StatOptions(duration_days=9)
-        link_stats = dynamic_links.get_link_stats(_MOCK_SHORT_URL, options, app=dltest.app)
+        link_stats = dynamic_links.get_link_stats(
+            MOCK_SHORT_URL, options, app=dynamic_links_test.app)
+        assert recorder[0].url.startswith("https://firebasedynamiclinks.googleapis.com")
+        assert (recorder[0].path_url ==
+                "/v1/https%3A%2F%2Ffake1.app.goo.gl%2FuQWc/linkStats?durationDays=9")
         assert isinstance(link_stats, dynamic_links.LinkStats)
-        assert isinstance(link_stats.event_stats[0], dynamic_links.EventStats)
-        assert link_stats.event_stats[0].platform == dynamic_links.PLATFORM_ANDROID
-        assert link_stats.event_stats[0].event == dynamic_links.EVENT_TYPE_CLICK
-        assert link_stats.event_stats[0].count == 123
-        assert len(link_stats.event_stats) == 7
-        assert repr(link_stats.event_stats[6]) == repr(
-            dynamic_links.EventStats(
-                platform=dynamic_links.PLATFORM_IOS,
-                event=dynamic_links.EVENT_TYPE_REDIRECT,
-                count=124))
-
-
-    def test_get_stats_error(self, dltest):
-        dltest._instrument_dynamic_links(payload=MOCK_GET_STATS_RESPONSE,
-                                         status=500)
+        for event_stat in link_stats.event_stats:
+            assert isinstance(event_stat, dynamic_links.EventStats)
+
+        compared_event_stats = get_link_stats_vals.comparison
+        assert len(compared_event_stats) == len(link_stats.event_stats)
+        for (direct, returned) in zip(compared_event_stats, link_stats.event_stats):
+            assert returned.platform == direct['platform']
+            assert returned.event == direct['event']
+            assert returned.count == direct['count']
+
+    def test_get_stats_error(self, dynamic_links_test):
+        err_body = json.dumps({'error': {
+            'status': 'INTERNAL_ERROR',
```

Comment:
What other status values are possible? I wonder if we should focus on that rather than the `code`. See how error handling is implemented in auth and fcm for example.